app/src/main/java/org/ea/sqrl/activites/UrlLoginActivity.java

```diff
@@ -231,4 +204,19 @@ protected void onStop() {
             hideProgressPopup();
         }
     }
+
+    private IdentitySelector configureIdentitySelector(SQRLStorage storage) {
```

Comment:
One thing I forgot to mention in the original message in this thread was that I came across an instance when resuming `UrlLoginActivity` where `mIdentitySelector` was null.  That precipitated `configureIdentitySelector()`.  I was considering just not having an instance variable at all, rather just building it as needed.  Considered (briefly) the idea of changing the `IdentitySelector` to a singleton pattern, but wasn't sure if that was appropriate (didn't think real hard about it).

Reply:
I am completely fine with what you did with the identity selector, I think the `configureIdentitySelector()` solution is appropriate to avoid ending up with an uninitialized selector, which I guess could have happend if onStop() was called upon leaving the activity without the system killing the app process and then the user navigates back to the activity.

I'm glad you caught this and will be more careful with this stuff in the future.